x-pack/plugins/security_solution/public/management/components/console/components/command_input/components/command_input_history.tsx

```diff
@@ -0,0 +1,160 @@
+    return list;
+  }, []);
+
+  const handleSelectableOnChange: EuiSelectableProps['onChange'] = useCallback(
+    (items: EuiSelectableOption[]) => {
+      optionWasSelected.current = true;
+
+      const selected = items.find((item) => item.checked === 'on');
+
+      dispatch({ type: 'updateInputPopoverState', payload: { show: undefined } });
+      dispatch({ type: 'updateInputPlaceholderState', payload: { placeholder: '' } });
+
+      if (selected) {
+        dispatch({ type: 'updateInputTextEnteredState', payload: { textEntered: selected.label } });
+      }
+
+      dispatch({ type: 'addFocusToKeyCapture' });
+    },
+    [dispatch]
+  );
+
+  const handleRenderOption = useCallback((option) => {
+    return <UserCommandInput input={option.label} />;
+  }, []);
+
+  // TODO:PT remove when https://github.com/elastic/eui/pull/5978 becomes available in kibana (task #4179)
+  const [checkObserverTrigger, setCheckObserverTrigger] = useState(1);
+
+  // TODO:PT remove when https://github.com/elastic/eui/pull/5978 becomes available in kibana (task #4179)
+  useEffect(() => {
+    let observer: MutationObserver;
+
+    if (optionListRef.current) {
+      observer = new MutationObserver((mutationList) => {
+        const focusedOption = mutationList.find((mutatedEl) =>
+          (mutatedEl.target as HTMLLIElement).classList.contains('euiSelectableListItem-isFocused')
+        );
+
+        if (focusedOption) {
+          dispatch({
+            type: 'updateInputPlaceholderState',
+            payload: {
+              placeholder: (focusedOption.target as HTMLLIElement).dataset.input as string,
+            },
+          });
+        }
+      });
+
+      observer.observe(optionListRef.current, {
+        attributes: true,
+        subtree: true,
+      });
+    } else {
+      setTimeout(() => {
+        if (isMounted) {
+          setCheckObserverTrigger(checkObserverTrigger + 1);
+        }
+      }, 5);
+    }
+
+    return () => {
+      if (observer) {
+        observer.disconnect();
+      }
+    };
+  }, [checkObserverTrigger, dispatch, isMounted]);
+
+  // When first loaded, clear out the current text entered, and when this component
+  // unloads, if no option from the history was selected, then set the prior text
+  // entered back
+  useEffect(() => {
+    dispatch({ type: 'updateInputTextEnteredState', payload: { textEntered: '' } });
+
+    return () => {
+      if (!optionWasSelected.current) {
+        dispatch({ type: 'updateInputTextEnteredState', payload: { textEntered: priorInputText } });
+        dispatch({ type: 'updateInputPlaceholderState', payload: { placeholder: '' } });
+      }
+    };
+  }, [dispatch, optionWasSelected, priorInputText]);
+
+  return (
+    <div>
+      <EuiSelectable
+        options={selectableHistoryOptions}
+        onChange={handleSelectableOnChange}
+        renderOption={handleRenderOption}
+        listProps={selectableListProps}
+        singleSelection={true}
```

Comment:
This should be `always` no? We can't enter multiple commands into the console anyway.

Reply:
I guess it is because it can be empty (no selection).

Reply:
@ashokaditya if `singleSelection` is set to `always` one item must always be "checked", even when the list is initially shown. In this usage, we don't care about the check marks on the items, so we never set them, just want the user to cilck/enter on a item

Reply:
Oh yes, I re-read what the props list on eui docs says, `always`: `can and must have only one selection`